src/stripe.common.ts

```diff
@@ -0,0 +1,481 @@
+  ) {
+    throw new Error(
+      'The `https` protocol must be used when sending requests to `*.stripe.com`'
+    );
+  }
+
+  const agent = props.httpAgent || null;
+
+  this._api = {
+    auth: null,
+    host: props.host || DEFAULT_HOST,
+    port: props.port || DEFAULT_PORT,
+    protocol: props.protocol || 'https',
+    basePath: DEFAULT_BASE_PATH,
+    version: props.apiVersion || DEFAULT_API_VERSION,
+    timeout: utils.validateInteger('timeout', props.timeout, DEFAULT_TIMEOUT),
+    maxNetworkRetries: utils.validateInteger(
+      'maxNetworkRetries',
+      props.maxNetworkRetries,
+      0
+    ),
+    agent: agent,
+    httpClient: props.httpClient || Stripe.createHttpClient(agent),
+    dev: false,
+    stripeAccount: props.stripeAccount || null,
+  };
+
+  const typescript = props.typescript || false;
+  if (typescript !== Stripe.USER_AGENT.typescript) {
+    // The mutation here is uncomfortable, but likely fastest;
+    // serializing the user agent involves shelling out to the system,
+    // and given some users may instantiate the library many times without switching between TS and non-TS,
+    // we only want to incur the performance hit when that actually happens.
+    Stripe.USER_AGENT.typescript = typescript;
+  }
+
+  if (props.appInfo) {
+    this._setAppInfo(props.appInfo);
+  }
+
+  this._prepResources();
+  this._setApiKey(key);
+
+  this.errors = _Error;
+  this.webhooks = require('./Webhooks');
+  this.webhooks._platformFunctions = Stripe._platformFunctions;
+
+  this._prevRequestMetrics = [];
+  this._enableTelemetry = props.telemetry !== false;
+
+  this._requestSender = new RequestSender(
+    this,
+    Stripe.StripeResource.MAX_BUFFERED_REQUEST_METRICS
+  );
+  // Expose StripeResource on the instance too
+  // @ts-ignore
+  this.StripeResource = Stripe.StripeResource;
+
+  this._platformFunctions = Stripe._platformFunctions;
+}
+
+Stripe.errors = _Error;
+Stripe.webhooks = require('./Webhooks');
+
+// @ts-ignore
+Stripe.createHttpClient = null;
+
+Stripe.createNodeHttpClient = (agent: http.Agent): void => {
+  throw new Error(
+    'Stripe: createNodeHttpClient() is not available in non-Node environments. When instantiating the Stripe client, please set the `httpClient` configuration option to `Stripe.createFetchHttpClient()`, or to your own implementation of `HttpClient`.'
+  );
+};
+
+/**
+ * Creates an HTTP client for issuing Stripe API requests which uses the Web
+ * Fetch API.
+ *
+ * A fetch function can optionally be passed in as a parameter. If none is
+ * passed, will default to the default `fetch` function in the global scope.
+ */
+Stripe.createFetchHttpClient = (fetchFn: typeof fetch): typeof HttpClient => {
+  const {FetchHttpClient} = require('./net/FetchHttpClient');
+  return new FetchHttpClient(fetchFn);
+};
+
+/**
+ * Create a CryptoProvider which uses the built-in Node crypto libraries for
+ * its crypto operations.
+ */
+Stripe.createNodeCryptoProvider = (): void => {
+  throw new Error(
```

Comment:
We'll need a corresponding fix in `webhooks` to create a different default crypto provider.

Reply:
Done in https://github.com/stripe/stripe-node/pull/1660/commits/ddb5e1562d6460a89f8d4102a2f63ded4bb91026